jellyfin-model/src/main/kotlin/org/jellyfin/apiclient/model/discovery/ServerVersion.kt

```diff
@@ -0,0 +1,46 @@
+package org.jellyfin.apiclient.model.discovery
+
+import kotlinx.serialization.Serializable
+import org.jellyfin.apiclient.model.discovery.ServerVersion.Companion.fromString
+
+/**
+ * Model to help with Jellyfin server versions.
+ * Use [fromString] to parse strings. The format is similar to SemVer.
+ */
+@Serializable
+data class ServerVersion(
+	val major: Int,
+	val minor: Int,
+	val patch: Int
+) {
+	operator fun compareTo(other: ServerVersion) = comparator.compare(this, other)
+
+	companion object {
+		private val comparator = compareBy<ServerVersion>(
+			{ it.major },
+			{ it.minor },
+			{ it.patch }
+		)
+
+		fun fromString(str: String): ServerVersion? {
+			try {
+				val parts = str
+					.split('.') // Split into major, minor and patch
+					.also { require(it.size == 3) } // Check if we found 3 parts
+					.take(3) // Take first 3 values in case the input is bad
```

Comment:
Is the `take` even needed with the require before?

Reply:
Nop this was a leftover from the previous implementation.